gobblin-utility/src/main/java/gobblin/util/SchedulerUtils.java

```diff
@@ -68,13 +75,22 @@ public boolean accept(File dir, String name) {
     }
   };
 
-  /**
-   * Load job configurations from job configuration files stored under the
-   * root job configuration file directory.
-   *
-   * @param properties Gobblin framework configuration properties
-   * @return a list of job configurations in the form of {@link java.util.Properties}
-   */
+  private static final PathFilter PROPERTIES_PATH_FILTER = new PathFilter() {
+    @Override
+    public boolean accept(Path path) {
+      String fileExtension = path.getName().substring(path.getName().lastIndexOf('.') + 1);
```

Comment:
Need not change but suggestion: 
`StringUtils.substringAfterLast(str, separator)` .. helps in such cases.

So this line would change to: 
`return JOB_PROPS_FILE_EXTENSION.equalsIgnoreCase(StringUtils.substringAfterLast(path.getName(), "."))` 

Can also use: `return path.getName().toLowerCase().endsWith("." + JOB_PROPS_FILE_EXTENSION)`